components/nimbus/src/tests/test_enrollment.rs

```diff
@@ -1595,6 +1633,367 @@ fn test_evolver_experiment_update_was_enrolled() -> Result<()> {
+    let exp4 = get_single_feature_experiment("exp4", "colliding", json!({"x": 2 }));
+
+    let all_experiments = [exp1, exp2, exp3.clone(), exp4.clone()];
+    let no_experiments: [Experiment; 0] = [];
+
+    let (enrollments, _) =
+        evolver.evolve_enrollment_recipes(true, &no_experiments, &all_experiments, &[])?;
+
+    let observed = map_features_by_feature_id(&enrollments, &all_experiments, &ids);
+    let expected = HashMap::from([
+        (
+            "colliding".to_string(),
+            EnrolledFeatureConfig::new("colliding", json!({"x": 1 }), "exp1", Some("control")),
+        ),
+        (
+            "coenrolling".to_string(),
+            EnrolledFeatureConfig::new(
+                "coenrolling",
+                json!({
+                    "a": 1, // from exp2
+                    "b": 3, // from exp3
+                    "c": 4, // from exp3
+                }),
+                "exp2+exp3",
+                None,
+            ),
+        ),
+    ]);
+    assert_eq!(observed, expected);
+
+    let experiments = [exp3, exp4];
+    let (enrollments, _) =
+        evolver.evolve_enrollment_recipes(true, &all_experiments, &experiments, &enrollments)?;
+
+    let observed = map_features_by_feature_id(&enrollments, &all_experiments, &ids);
+    let expected = HashMap::from([
+        (
+            "colliding".to_string(),
+            EnrolledFeatureConfig::new("colliding", json!({"x": 2 }), "exp4", Some("control")),
+        ),
+        (
+            "coenrolling".to_string(),
+            EnrolledFeatureConfig::new(
+                "coenrolling",
+                json!({
+                    "b": 3, // from exp3
+                    "c": 4, // from exp3
+                }),
+                "exp3",
+                Some("control"),
+            ),
+        ),
+    ]);
+    assert_eq!(observed, expected);
+    Ok(())
+}
+
+#[test]
+fn test_evolve_enrollments_with_coenrolling_multi_features() -> Result<()> {
+    let _ = env_logger::try_init();
+    let (nimbus_id, app_ctx, aru) = local_ctx();
+    let th = app_ctx.into();
+    let ids = HashSet::from(["coenrolling"]);
+    let evolver = EnrollmentsEvolver::new(&nimbus_id, &aru, &th, &ids);
+
+    let exp1 = get_multi_feature_experiment(
+        "exp1",
+        "colliding",
+        json!({"x": 1 }),
+        "coenrolling",
+        json!({ "a": 1, "b": 2 }),
+    );
+    let exp2 = get_single_feature_experiment("exp2", "coenrolling", json!({ "b": 3, "c": 4 }));
+    let exp3 = get_multi_feature_experiment(
+        "exp3",
+        "colliding",
+        json!({"y": 1 }),
+        "coenrolling",
+        json!({ "c": 5, "d": 6 }),
+    );
+    let exp4 = get_multi_feature_experiment(
+        "exp4",
+        "another",
+        json!({"p": 1 }),
+        "coenrolling",
+        json!({ "d": 7, "e": 8 }),
+    );
+
+    let all_experiments = [exp1, exp2, exp3.clone(), exp4.clone()];
```

Comment:
> Thinking about the case where two experiments set different values for the same feature variable — wondering if each client could end up seeing different things as a result of that

As an example of this, what happens if we swap the order of these experiments around? Is there something somewhere that re-orders them based on something deterministic? 

Additionally, if you apply say experiments 1 and 2, then later apply experiments 1, 2, 3, and 4 do we end up with the same result as just applying all four of them at once? That might be more for a `nimbus_client` integration-style test, but I feel like it could be worthwhile to have.

Reply:
I do want to say I recognize that in practice this scenario _should_ never happen, as coenrolling features should be highly configurable and key-value based (so key collisions are unlikely at best), but considering we don't have fine-tuned control over how users/devs are able to use coenrolling I think it'd be wise of us to at least know what we can expect if an app ends up in that scenario.

Do we have any plans to update Experimenter to identify targeting/key collisions across experiments for coenrolling features?

Reply:
Also also do we need to consider salting bucket sampling for experiments targeting only coenrolling features?

Reply:
> Also also do we need to consider salting bucket sampling for experiments targeting only coenrolling features?

Could you elaborate? perhaps in a document or meeting.

Reply:
Never mind, a quick look through the available experiments on Experimenter indicates to me we already do what I was referring to with namespaces right now